src/driver/op.rs

```diff
@@ -37,28 +48,33 @@ pub(crate) enum Lifecycle {
     /// must be passed to the driver and held until the operation completes.
     Ignored(Box<dyn std::any::Any>),
 
-    /// The operation has completed.
+    /// The operation has completed with a single result
```

Comment:
The single Op case is unchanged. We don't pay any slab lookup overhead for single completions